lib/chai/interface/assert.js

```diff
@@ -1232,66 +1232,78 @@ module.exports = function (chai, util) {
   }
 
   /**
-   * ### .throws(function, [constructor/string/regexp], [string/regexp], [message])
+   * ### .throws(fn, [errorLike/string/regexp], [string/regexp], [message])
    *
-   * Asserts that `function` will throw an error that is an instance of
-   * `constructor`, or alternately that it will throw an error with message
-   * matching `regexp`.
+   * Asserts that `fn` will throw an error that is an instance of `errorLike`
+   * (if `errorLike` is an `Error` constructor) or if it is the same instance as `errorLike`
+   * (if `errorLike` is an `Error` instance). This also asserts that the error thrown
+   * will have a message matching `errMsgMatcher`.
    *
    *     assert.throws(fn, 'function throws a reference error');
    *     assert.throws(fn, /function throws a reference error/);
    *     assert.throws(fn, ReferenceError);
-   *     assert.throws(fn, ReferenceError, 'function throws a reference error');
-   *     assert.throws(fn, ReferenceError, /function throws a reference error/);
+   *     assert.throws(fn, new ReferenceError());
+   *     assert.throws(fn, ReferenceError, 'Error thrown must be a ReferenceError and have this msg');
+   *     assert.throws(fn, new ReferenceError(), 'Error thrown must be the same ReferenceError instance and have this msg');
+   *     assert.throws(fn, ReferenceError, /Error thrown must be a ReferenceError and match this/);
+   *     assert.throws(fn, new ReferenceError(), /Error thrown must be the same ReferenceError instance and match this/);
```

Comment:
I'm torn about the examples in 1245, 1247, and 1249. Although I understand they're merely meant to demonstrate what happens when an instance is provided as the second argument instead of a constructor, I'm not crazy about the fact that these examples would fail 100% of the time no matter what, since there's no way `fn` could possibly throw the same Error instance that's being instantiated in this way. I'd feel fuzzier if these examples demonstrated more realistic usage.

One idea is to change `new ReferenceError()` in the examples to `errInstance`.